app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java

```diff
@@ -0,0 +1,102 @@
+package cs5150athletetracking.com.athletetracking;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.io.*;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.net.URLEncoder;
+import java.util.Iterator;
+
+import org.json.JSONObject;
+
+
+/**
+ * Created by mag94 on 10/19/15.
+ */
+
+public class AsyncUploader extends AsyncTask<JSONObject, String, Integer> implements ResultCallable {
```

Comment:
I think it might be best to just have a setter for a callback and then later on call it (if the callback isn't null)